crates/solvers/src/boundary/balancer.rs

```diff
@@ -0,0 +1,93 @@
+//! Balancer boundary module. This contains encoding logic for the Balancer V2
+//! vault contract. The reason this was moved to a boundary module is that it
+//! currently requires `web3` as dependencies (not to mention needing to
+//! instantiate a "dummy" web3 instance) for ABI encoding, which should **not**
+//! be necessary at all. The swap parameter types are here instead of the domain
+//! as they should be generated with the contract bindings (and are more like
+//! DTOs than domain objects anyway).
```

Comment:
Putting the types here rather than the domain was indeed the right decision. However, if the domain ever needed to access them, that would probably be a good indicator that this needs to be refactored and they should be moved into the domain. But currently AFAIU they're only accessed by `infra` so this is totally fine. :+1: